### ISSUE TITLE
[timebox: 1w] "ckan in a box" docker-compose file for testing ckan extensions

### ISSUE THREAD

**Issue by adborden** [GSA/data.gov#2449]
## User Story

In order to quickly and easily spin up a CKAN instance for local development and CI testing, data.gov developers want a drop-in docker-compose file that can be used for testing CKAN extensions and running CI tests on extension forks.

## Acceptance Criteria
- [ ] GIVEN I have a CKAN extension that requires a CKAN setup for testing
  WHEN I drop in the docker-compose.yml
  AND I run `docker-compose run ckan <test-command>`
  THEN the extension tests are executed
- [ ] GIVEN I have a CKAN extension that requires a CKAN setup for testing
  WHEN I drop in the docker-compose.yml
  AND I set CKAN_VERSION=2.9
  AND I run `docker-compose run ckan <test-command>`
  THEN the CKAN extension tests are executed in an environment with the correct version of CKAN

## Background
There are several examples of CKAN docker setups:
- https://github.com/okfn/docker-ckan
- https://github.com/keitaroinc/docker-ckan
- https://github.com/GSA/catalog.data.gov
- https://github.com/GSA/inventory-app
- https://github.com/GSA/catalog-app

However, most of these are not self contained. They build containers at runtime and require additional files. If you were to use them in an existing extension, you'd have many files to copy. Some of these containers have no documentation and they require some configuration. Some of these images are very large.

## Security Considerations ([required](https://nvd.nist.gov/800-53/Rev4/control/CM-4))
[comment]: # "Our SSP says 'The Data.gov team ensures security implications are considered as part of the agile requirements refinement process by including a section in the issue template used as a basis for new work.' so please don't remove this section without care."
None, these containers and docker-compose files would only be used in development and should follow existing development best practices.

## Sketch
- Images should be pre-built where possible
- Usage should be straight forward: copy docker-compose.yml, copy a CI config for CKAN matrix build, add a command for installing extension dependencies and running tests.

**Comment by jbrown-xentity** [comment]
We believe this would not be much work; but the OKFN image build speed would be greatly increased by moving to python3.x series images...

**Comment by hkdctol** [comment]
Work involved is not worth the improvement we would get back, so staying with current architecture and closing ticket.

**Comment by nickumia-reisys** [comment]
As a summary:
- `catalog` was converted from using the `okfn` image to the `python` base image for performance improvements and more similar testing environment to production
  - https://github.com/GSA/catalog.data.gov/pull/612
- `inventory` was left untouched and is still running the `okfn` image
  - https://github.com/GSA/inventory-app/blob/main/Dockerfile
- All current extensions use the `okfn` image as well.

It is hard to consolidate our development environment because our codebases are so vast.  I think I prefer the `python` base image over the `alpine` one that okfn uses; but it's not a strong enough preference to inspire a change in how we are doing things.  There is not a clear better path.

**Comment by btylerburton** [comment]
@nickumia-reisys given what we were talking about yesterday with playwright & pytest not being compatible with the Alpine image, what do you think the level of effort would be to those catalog changes to an extension?

**Comment by nickumia-reisys** [comment]
I did try to port it.  The long story short is it would be more work that expected:
- We wanted to have minimal code duplication across repos.  With the build time vs. runtime tradeoffs, there wasn't a great way of inheriting catalog's code and pulling in extension-specific changes/configuration.
- It was more bloated that I wanted it to be.

Our current `okfn` inheritance is pretty broken in that, per extension, there are various features that don't work the same way and it's very hard to keep track of all of the minor differences.  You saw this when we were fixing the tests for `metric_dashboard`, there were random things that we needed to configure that we're explicitly referenced in our other repos.
